### PR TITLE
[llm-load-test] Switch to v2

### PR DESCRIPTION
**Pull request by dagrayvid**

### CODE REVIEW

File: projects/kserve/testing/config.yaml [hunk 1]
```diff
@@ -96,7 +96,7 @@ ci_presets:
 
   matbenchmark:
     tests.e2e.matbenchmark.enabled: true
-    tests.e2e.llm_load_test.threads: [8, 16, 32]
+    tests.e2e.llm_load_test.: [8, 16, 32]
```
Comment (kpouget) [inline projects/kserve/testing/config.yaml:99]: `concurrency` is missing here

File: .gitmodules [hunk 2]
```diff
@@ -4,3 +4,4 @@
 [submodule "subprojects/llm-load-test"]
 	path = subprojects/llm-load-test
 	url = https://github.com/openshift-psap/llm-load-test
+	branch = main
```
Comment (kpouget) [inline .gitmodules:7]: I'm not sure if this changes the fact that `subprojects/llm-load-test` is pinned to a particular commit of `openshift-psap/llm-load-test` or not?
(I'd like to keep the pinning)

File: projects/llm_load_test/roles/llm_load_test_run/tasks/main.yml [hunk 3]
```diff
@@ -36,31 +33,13 @@
...
-      cat '{{ artifact_extra_logs_dir }}'/output/ghz-multiplexed-results.orig \
-        | jq '.[].details[].response |= fromjson' \
-        > "${output_file}"
-      echo "Results have been save in '${output_file}'"
 
   - name: Ensure that some content has been generated
     shell:
       set -o pipefail;
-      cat {{ artifact_extra_logs_dir }}/output/ghz-multiplexed-results*.json | jq .[] | grep . --quiet
+      cat {{ artifact_extra_logs_dir }}/output/outpout.json | grep . --quiet
```
Comment (kpouget) [inline projects/llm_load_test/roles/llm_load_test_run/tasks/main.yml:42]:
```suggestion
      cat {{ artifact_extra_logs_dir }}/output/output.json | grep . --quiet
```

File: projects/kserve/testing/test_e2e.py [hunk 4]
```diff
@@ -557,7 +557,7 @@ def test_consolidated_model(consolidated_model, namespace=None):
         protos_path=protos_path.absolute(),
         call=llm_config["call"],
         llm_path=llm_config["src_path"],
-        threads=llm_config["threads"],
+        concurrency=llm_config["concurrency"],
```
Comment (kpouget) [inline projects/kserve/testing/test_e2e.py:560]: I understand that no proto is required anymore with v2,
could you remove the code generating them in this file, as well as the proto files?
(we'll keep the roles, they might be useful again in the future, who knows)

**Review by kpouget — CHANGES_REQUESTED** [review]
I made a few minor comments
we'll have to run the tests (single and multi model), and the kserve `visualizations` [parser](https://github.com/openshift-psap/topsail/blob/main/projects/kserve/visualizations/kserve-llm/store/parsers.py#L152) will have to be updated. Maybe the plotting files as well, as I think I currently store untouched the llm-load-test output.

Could you also try to move `subprojects/llm-load-test` into `projects/llm-load-test/llm_load_test`? it didn't work right away when I did the refactoring with the `projects` directory, because Prow couldn't build the image. I didn't go in depth into the issue.

**Comment by dagrayvid** [comment]
Okay, this last test succeeded to the extent that we want, we got [output from llm-load-test-v2](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/193/pull-ci-openshift-psap-topsail-main-rhoai-light/1761118573322833920/artifacts/light/test/artifacts/000__local_ci__run_multi_e2e_perf_test/artifacts/ci-pod-0/001__flan-t5-small-cpu/006__llm_load_test__run/output/output.json), we just need to disable the results parsing for now.

**Comment by dagrayvid** [comment]
@kpouget test is green

**Comment by kpouget** [comment]
thanks @dagrayvid, looks good, I'll merge it.

The visualization (now disabled) fails because of [this error](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/193/pull-ci-openshift-psap-topsail-main-rhoai-light/1761118573322833920/artifacts/light/test/artifacts/003__plots/000__projects.kserve.visualizations.kserve-llm_plots/0_matbench_parse.log).

Here is [a sample](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/193/pull-ci-openshift-psap-topsail-main-rhoai-light/1761118573322833920/artifacts/light/test/artifacts/000__local_ci__run_multi_e2e_perf_test/artifacts/ci-pod-0/001__flan-t5-small-cpu/006__llm_load_test__run/output/output.json) of the next output file of `llm-load-test`.

**Review by kpouget — APPROVED** [review]

**Pull request merged**